_drafts/2020-07-27-data_science_plotting_notebook_template.md

```diff
@@ -0,0 +1,131 @@
+
+I recently released my [Jupyter Notebook Template Library][library]. Its goal
+is to accelerate your data science projects without having to to spend hours
+poking through old notebooks to find handy code snippets. In this post I dive
+into the plotting notebook to show you what it can do.
+
+[library]: https://github.com/agude/Jupyter-Notebook-Template-Library
+
+[nb_post]: {% post_url 2016-10-17-jupyter_not_for_development %}
+
+## The Plotting Notebook
+
+Visualizing your data is a critical step in understanding your data, and so it
+is appropriate that the [**first notebook in the library**][plotting_nb] helps
+with making beautiful plots.
+
+[plotting_nb]: https://github.com/agude/Jupyter-Notebook-Template-Library/blob/d6cda39c388154cb8f4073e669efff109c743a99/notebooks/basic-plotting-template.ipynb
+
+The notebook begins with a bunch of boilerplate to set some metadata that is
+saved to the files and change some defaults like the figure size and
+resolution, font size, and legend frame. After that there are a few helpful
+functions which I will discuss below.
+
+### Draw Bands
+
+One of my favorite functions is `draw_bands()`. It draws a set of colored
+bands on the background of the plot based on the axis tick locations.
+
+When called with just the axis, like `draw_bands(ax)`, it produces this:
+
+[![A plot showing the default grey bands.][bands_plot]][bands_plot]
+
+[bands_plot]: {{ file_dir }}/bands.svg
+
+But you can also customize the color using `draw_bands(ax, color="orange",
+alpha=0.05)`, which produces:
+
+[![A plot showing the orange bands.][orange_bands_plot]][orange_bands_plot]
+
+[orange_bands_plot]: {{ file_dir }}/orange_bands.svg
+
+These bands are a subtle way of indicating where on the X-axis a point lies,
+which is especially useful when plotting a time series. I use them often. Here
+are some examples:
+
+- [**Discussing my sons' language development**][language_post] to highlight each month.
+- [**Plotting the progression of the cycling hour record**][hour_post] to show each decade.  
+- [**Exploring when cyclists are involved in traffic accidents**][bike_post] to highlight the seasonality.
+
+[language_post]: {% post_url 2020-02-10-my_sons_language_development_comparison %}#development
+[hour_post]: {% post_url 2019-07-09-hour_record_plot_improvements %}#improvements
+[bike_post]: {% post_url 2019-02-20-switrs_bicycle_crashes_by_date %}#day-by-day
+
+### Draw Legends
+
+I like minimal, but informative, legends. Color alone is often enough to
+differentiate lines or points on simple plots, so I wrote a function to change
+the color of the legend text to match the line, called
+`draw_colored_legend()`. It produces a legend like on this plot:
+
+[![A plot showing my colored legend.][legend_plot]][legend_plot]
+
+[legend_plot]: {{ file_dir }}/legend.svg
+
+This function is a little [brittle][brittle]; it works well for scatter plots,
+but fails for some other matplotlib objects. I plan to make it more robust in
```

Comment:
Capitalize Matplotlib?

Reply:
Also, do you have an open issue to clean this up, or a todo list in your readme, or something?

And will you remember to come clean up this promise after you make the code more robust?

Reply:
https://github.com/agude/Jupyter-Notebook-Template-Library/issues/3#issuecomment-663923874